src/App.js

```diff
@@ -49,16 +51,26 @@ export default class App extends React.Component {
 		this.setInStorage('theme', theme);
 	};
 	addToPhotos = (photo) => {
-		let savedPhotos = [...this.state.savedPhotos, photo];
-		this.setState({ savedPhotos: savedPhotos });
-		this.setInStorage('savedPhotos', savedPhotos);
+		if (
+			this.state.savedPhotos.filter((saved) => saved.id === photo.id).length ===
+			0
+		) {
+			let savedPhotos = [...this.state.savedPhotos, photo];
+			this.setState({ savedPhotos: savedPhotos });
+			this.setInStorage('savedPhotos', savedPhotos);
+		}
 	};
 	addToCollections = (collection) => {
-		let savedCollections = [...this.state.savedCollections, collection];
-		this.setState({
-			savedCollections: savedCollections,
-		});
-		this.setInStorage('savedCollections', savedCollections);
+		if (
+			this.state.savedCollections.filter((saved) => saved.id === collection.id)
+				.length === 0
+		) {
+			let savedCollections = [...this.state.savedCollections, collection];
+			this.setState({
+				savedCollections: savedCollections,
+			});
+			this.setInStorage('savedCollections', savedCollections);
+		}
```

Comment:
Check if the Collection already exists in savedCollections

Reply:
same here